test/e2e/util.go

```diff
@@ -509,6 +509,44 @@ func deleteTestingNS(c *client.Client) error {
 	return fmt.Errorf("Waiting for terminating namespaces to be deleted timed out")
 }
 
+// deleteNS deletes the provided namespaces, waits for it to be completely deleted, and then checks
+// whether there are any pods remaining in a non-terminating state.
+func deleteNS(c *client.Client, namespace string) error {
+	if err := c.Namespaces().Delete(namespace); err != nil {
+		return err
+	}
+
+	err := wait.Poll(1*time.Second, 2*time.Minute, func() (bool, error) {
+		if _, err := c.Namespaces().Get(namespace); err != nil {
+			if apierrs.IsNotFound(err) {
+				return true, nil
+			}
+			Logf("Error while waiting for namespace to be terminated: %v", err)
+			return false, nil
+		}
+		return false, nil
+	})
+
+	// check for pods that were not deleted
+	if pods, perr := c.Pods(namespace).List(labels.Everything(), fields.Everything()); perr == nil {
+		deleted := true
+		for _, pod := range pods.Items {
+			if pod.DeletionTimestamp == nil {
+				deleted = false
+			}
+			Logf("Pod %s %s on node %s has deletion timestamp %s", namespace, pod.Name, pod.Spec.NodeName, pod.DeletionTimestamp)
+			if !deleted {
+				return fmt.Errorf("some pods in namespace %s were not deleted by the node controller: %v", pod.Name, err)
```

Comment:
maybe build up a list of pods not deleted, rather than only reporting the first one not deleted?